pkg/operator/ceph/nvmeof_recoverer/nvmeofstorage/controller.go

```diff
@@ -105,5 +109,67 @@ func add(mgr manager.Manager, r reconcile.Reconciler) error {
 func (r *ReconcileNvmeOfStorage) Reconcile(context context.Context, request reconcile.Request) (reconcile.Result, error) {
 	logger.Debugf("reconciling NvmeOfStorage. Request.Namespace: %s, Request.Name: %s", request.Namespace, request.Name)
 
-	return reporting.ReportReconcileResult(logger, r.recorder, request, nil, reconcile.Result{}, nil)
+	if strings.Contains(request.Name, "nvmeofstorage") {
+		// Fetch the NvmeOfStorage CRD object
+		err := r.client.Get(r.opManagerContext, request.NamespacedName, r.nvmeOfStorage)
+		if err != nil {
+			logger.Errorf("unable to fetch NvmeOfStorage, err: %v", err)
+			return reconcile.Result{}, err
+		}
+
+		// Retrieve the nvmeofstorage CR to update the CRUSH map
+		for i := range r.nvmeOfStorage.Spec.Devices {
+			device := &r.nvmeOfStorage.Spec.Devices[i]
+			// Get OSD pods with label "app=rook-ceph-osd"
+			var osdID, clusterName string
+			pods, err := r.context.Clientset.CoreV1().Pods(request.Namespace).List(context, metav1.ListOptions{
+				LabelSelector: "app=rook-ceph-osd",
+			})
+			if err != nil {
+				panic(err)
+			}
+			// Find the osd id and cluster name for the fabric device listed in the nvmeofstorage CR
+			for _, pod := range pods.Items {
+				if pod.Spec.NodeName == device.AttachedNode {
+					for _, container := range pod.Spec.Containers {
+						for _, env := range container.Env {
+							if env.Name == "ROOK_BLOCK_PATH" && env.Value == device.DeviceName {
+								osdID = pod.Labels["ceph-osd-id"]
+								clusterName = pod.Labels["rook_cluster"]
+								break
+							}
+						}
+					}
+				}
+			}
+
+			// Update CRUSH map for OSD relocation to fabric failure domain
+			fabricHost := "fabric-host-" + r.nvmeOfStorage.Spec.Name
+			clusterInfo := cephclient.AdminClusterInfo(context, request.Namespace, clusterName)
+			cmd := []string{"osd", "crush", "move", "osd." + osdID, "host=" + fabricHost}
+			exec := cephclient.NewCephCommand(r.context, clusterInfo, cmd)
+			exec.JsonOutput = true
+			buf, err := exec.Run()
+			if err != nil {
+				logger.Error(err, "Failed to move osd", "osdID", osdID, "srcHost", device.AttachedNode,
+					"destHost", fabricHost, "result", string(buf))
+				panic(err)
+			}
+			logger.Debugf("Successfully updated CRUSH Map. osdID: %s, srcHost: %s, destHost: %s",
+				osdID, device.AttachedNode, fabricHost)
+
+			// Update the osdID assignments for devices listed in the NvmeOfStorage CR
+			device.OsdID = osdID
```

Comment:
Why do we need to keep osdID?

Reply:
Upon further consideration, we might not need to keep it.
It was originally used to track the device path when an OSD Pod faults, since only the osdID(pod name) is provided to the reconciler.
However, we can also directly obtain the device path from the pod's information using the Kubernetes API.

I will proceed with removing it.